### ISSUE TITLE
Undo last action improvement. 

### ISSUE THREAD

**Issue by kkrawczyk123** [benetech/MathShare#358]
Currently, when the user chooses to undo last action button after adding, deleting or editing steps - content of the of "undone" step is visible in my work area. Adding and deleting step concerns always the last added step so it is not such a big deal, but edit doesn't have to. 
As an example: I edited step 2 and undo the edition, this step is displayed in my work content as on attached screenshot, but I am a bit further with the progress of my work - I am on step 5. 
![screenshot from 2018-08-01 15-18-51](https://user-images.githubusercontent.com/33342488/43524198-ec7f2be0-959e-11e8-844b-d220aa8e9b04.png)
I think it can be confusing for users and I would like to suggest that after reversing that action in my work area there should be the content of the last step and empty comment field. The exact behavior as after updating a step, screenshot attached:
![screenshot from 2018-08-01 15-17-27](https://user-images.githubusercontent.com/33342488/43524399-83c7ad7e-959f-11e8-809b-e306e2165a2e.png)
It would be much easier for user to continue his work, @johnhbenetech what do you think about that solution?

**Comment by johnhbenetech** [comment]
that sounds like a good suggestion @kkrawczyk123 ! we can put this is the back log.

**Comment by kkrawczyk123** [comment]
@afijal, please proceed.

**Comment by kkrawczyk123** [comment]
After implementation of https://github.com/benetech/MathShare/issues/365 undo button is visible all the time, it was visible only after action before. It would be good to hide it when there is no action.

**Comment by kkrawczyk123** [comment]
@afijal, when update step which has a cleanup step and undo the update - edition is reversed but cleanup step is disappearing. It would be good to fix it as well.

**Comment by kkrawczyk123** [comment]
The improvement described in the issue has been added to the app and bug connected to disappearing cleanup step has been fixed too. There are still some other issues connected to undo action but as they are not strictly connected to that ticket so I decided to report it as separate issue in https://github.com/benetech/MathShare/issues/490
